### DISCUSSION TITLE
External DNS, Wildcard DNS, Gateway Routes, and feedback!

### DISCUSSION THREAD

**Discussion by mitchross** [vehagn/homelab#20] [Ideas]
I left a comment on your blog post, but I cant seem to find the comment section, I really like your setup.  I had been following https://github.com/acelinkio/argocd-homelab for my own setup. Its very clean ,very simple with lots of power. I think you can significantly cut down on some boiler plate code you have around http routes and cilium.

I was seeking out your repo for your use of application set and separation of plain k8s manifest from argo. Many home lab projects tightly couple their deployments to argo, where they cant deploy without argo. Thats not very enterprise ready, albeit it is a home lab, i do think separation of concerns is still a good idea.

I think you should check out that repo above to finish out your todo list and maybe simplify your base a bit. I do reckon that I was using https://gitlab.com/vehagn/mini-homelab as a guide so maybe this repo I need to study more.. 

Anywho keep up the good work and motivation because I think you have one of the best setups. I think theres too much ansible in other project repos that add way too much complexity and end up being very difficult to fork and follow along. Your guides on your blog post are good enough to bootstrap and dont really warrant Ansible.  

I had previously left a comment about declarative argo issues on your blog post and I do still think you could probably leverage argo more to setup your whole cluster, see the setup guide to draw inspiration  https://github.com/mitchross/argocd-homelab/blob/2841158f01d93e4866faa78c2f93b16317ea293a/docs/setup.md

keep up the good work, im following this repo!

**Comment by vehagn** [comment]
Thank you for the kind words! I'll check the sources you link for inspiration as there's always room for improvement!

I had a slight brain-fart some weeks ago and deleted all my volumes without having a proper backup first, so I apologise for losing your comment! I also don't have a backup solution for the comments section as I do for the blog itself, so sometimes it's unavailable when I try stuff out and break something.
